benchmarks/scripts/sol.py

```diff
@@ -106,22 +96,37 @@ def plot_sol(medians, box):
         ax = sns.boxenplot(data=medians, x='alg', y='bw', hue='hue')
     else:
         ax = sns.barplot(data=medians, x='alg', y='bw', hue='hue', errorbar=lambda x: (x.min(), x.max()))
-    for container in ax.containers:
-        ax.bar_label(container, fmt='%.1f')
-    ax.set_xticklabels(ax.get_xticklabels(), rotation=15, rotation_mode='anchor', ha='right')
+        ax.bar_label(ax.containers[0], fmt='%.1f')
+        for container in ax.containers[1:]:
+            labels = [f'{c:.1f}\n({(c/f)*100:.0f}%)' for f, c in zip(ax.containers[0].datavalues, container.datavalues)]
+            ax.bar_label(container, labels=labels)
+
+    ax.legend(title=None)
+    ax.set_xlabel('Algorithm')
+    ax.set_ylabel('Bandwidth (%SOL)')
+    ax.set_xticklabels(ax.get_xticklabels(), rotation=30, rotation_mode='anchor', ha='right')
     plt.show()
 
+def print_speedup(medians):
+    m = medians.groupby(['alg', 'hue'], sort=False).mean()
+    m['speedup'] = (m['bw'] / m.groupby(['alg'])['bw'].transform('first'))
+    print('# Speedups:')
+    print()
+    print(m.drop(columns='bw').sort_values(by='speedup', ascending=False).to_markdown())
```

Comment:
This feels a bit misplaced here, but it's great to have the speedup's from the plot as a copyable table as well., since `compare.py` works on a much more detailed level.